api/src/main/java/org/apache/iceberg/io/FileIO.java

```diff
@@ -49,25 +49,25 @@ default InputFile newInputFile(String path, long length) {
   default InputFile newInputFile(DataFile file) {
     Preconditions.checkArgument(
         file.keyMetadata() == null,
-        "Cannot decrypt data file: {} (use EncryptingFileIO)",
+        "Cannot decrypt data file: %s (use EncryptingFileIO)",
         file.path());
-    return newInputFile(file.path().toString());
+    return newInputFile(file.path().toString(), file.fileSizeInBytes());
```

Comment:
actually not quite right, since the default implementation of newInputfile with a path actually ignores the path.

Reply:
no I confused myself, actually the default is fine. It's up to FileIO implementations to override newInputfile with a size and keep track of the size. That's what S3FileIO and others do.